contracts/token/ERC20/EncryptedERC20.sol

```diff
@@ -1,201 +1,182 @@
 // SPDX-License-Identifier: BSD-3-Clause-Clear
-
-pragma solidity ^0.8.20;
+pragma solidity ^0.8.24;
 
 import "fhevm/lib/TFHE.sol";
-import "fhevm/abstracts/Reencrypt.sol";
-import "../../utils/EncryptedErrors.sol";
+import { IEncryptedERC20 } from "./IEncryptedERC20.sol";
 
-abstract contract EncryptedERC20 is Reencrypt, EncryptedErrors {
-    enum ErrorCodes {
-        NO_ERROR,
-        UNSUFFICIENT_BALANCE,
-        UNSUFFICIENT_APPROVAL
-    }
+/**
+ * @title       EncryptedERC20
+ */
+abstract contract EncryptedERC20 is IEncryptedERC20 {
+    /// @notice Number of decimals.
+    uint8 internal constant _DECIMALS = 6;
```

Comment:
please use consistant casing, and imo keep same formatting and visibility that the one used in OpenZepelin, meaning remove this `_DECIMALS`private constant, and replace it with this public virtual view function: `    function decimals() public view virtual returns (uint8) {
        return 6;
    }`